app/services/dollar_parser.rb

```diff
@@ -1,5 +1,16 @@
-class DollarParser < BaseParser
-  def separator
-    '$'
+class DollarParser
+  def initialize(raw_data)
+    @parser = CsvParser.new(raw_data, ' $ ')
+  end
+
+  def people
+    @parser.rows.map do |row|
+      Person.new(
+        city: row['city'],
+        birthdate: Date.strptime(row['birthdate'], '%d-%m-%Y'),
```

Comment:
Explicit is better than implicit.
I just found out `Date.parse` is super smart, but it might fail you some times, like it did with the `.` date from the pipe formatter.